src/acsets/amr_conversion/stockflow.py

```diff
@@ -0,0 +1,187 @@
+
+    for idx, flow in enumerate(flows):
+        flow_id = 'flow' + str(flow['_id'])
+        flow_name = flow['fname']
+        upstream_stock = stocks[int(flow['u']) - 1]['sname']
+        downstream_stock = stocks[int(flow['d']) - 1]['sname']
+
+        expression_str = flow['ϕf'].replace('p.', '').replace('u.', '')
+
+        flow_dict = {}
+        flow_dict['id'] = flow_id
+        flow_dict['name'] = flow_name
+        flow_dict['upstream_stock'] = upstream_stock
+        flow_dict['downstream_stock'] = downstream_stock
+        flow_dict['rate_expression'] = expression_str
+
+        params.extend(re.findall(r'p\.([^()*+-/ ]+)', flow['ϕf']))
+
+        params_stock_flow_map[flow_id] = []
+
+        params_stock_flow_map[flow_id].extend(
+            re.findall(r'p\.([^()*+-/ ]+)', flow['ϕf']))
+        params_stock_flow_map[flow_id].extend(
+            re.findall(r'u\.([^()*+-/ ]+)', flow['ϕf']))
+
+        flows_list.append(flow_dict)
+
+    for param in params:
+        param_dict = {}
+        param_name = 'p_' + param
+        param_dict['id'] = param_name
+        param_dict['name'] = param_name
+        param_dict['value'] = 0.0
+        params_list.append(param_dict)
+
+        auxiliary_dict = {}
+        auxiliary_dict['id'] = param
+        auxiliary_dict['name'] = param
+        auxiliary_dict['expression'] = param_name
+        auxiliaries_list.append(auxiliary_dict)
+
+    idx = 0
+    for flow_id, stock_param_list in params_stock_flow_map.items():
+        for stock_or_param in stock_param_list:
+            link_dict = {'id': 'link' + str(idx + 1)}
+            link_dict['source'] = stock_or_param
+            link_dict['target'] = flow_id
+
+            links_list.append(link_dict)
+            idx += 1
+
+    return {
+        'header': {
+            'name': '',
+            'schema': '',
+            'description': '',
+            'schema_name': 'stockflow',
+            'model_version': '0.1',
+        },
+        'model': {
+            'flows': flows_list,
+            'stocks': stocks_list,
+            'auxiliaries': auxiliaries_list,
+            'links': links_list
+        },
+        'semantics': {'ode': {
+            'parameters': params_list,
+            'initials': initials_list,
+            'observables': [],
+            'time': None
+        }},
+        'metadata': None
+    }
+
+
+def amr_to_acset(amr):
+    """
+    A method that takes in a stock and flow JSON amr dictionary
+    object and outputs an equivalent stock and flow
+    JSON acset dictinary object
+    """
+    flows = amr['model']['flows']
+    stocks = amr['model']['stocks']
+    links = amr['model']['links']
+
+    flows_list = []
+    stocks_list = []
+    links_list = []
+
+    symbols = {}
+    stocks_mapping = {}
+    for parameter in amr['semantics']['ode']['parameters']:
+        if parameter['id'].startswith('p_'):
+            symbols[parameter['id'][2:]] = sympy.Symbol(
+                'p.' + parameter['id'][2:])
```

Comment:
The whole p. and u. thing are implementation details in julia that the state variable of a dynamical system is called u and the parameters are p. Do we want to cannonicalize that in the standard?

Reply:
@jpfairbanks could you confirm if the `u.` and `p.` are expected in the ACSet or if we should remove this assumption?